plugins/plugin-java/che-plugin-java-ext-debugger-java-client/src/main/java/org/eclipse/che/ide/ext/java/jdi/client/debug/JavaDebugger.java

```diff
@@ -0,0 +1,735 @@
+
+        for (String sourceFolder : sourceFolders) {
+            filePaths.add(sourceFolder + pathSuffix);
+        }
+        filePaths.add(location.getClassName());
+
+        return filePaths;
+    }
+
+    private void startCheckingEvents() {
+        debuggerEventsChannel = JavaRuntimeExtension.EVENTS_CHANNEL + javaDebuggerInfo.getId();
+        try {
+            messageBus.subscribe(debuggerEventsChannel, debuggerEventsHandler);
+        } catch (WebSocketException e) {
+            Log.error(DebuggerPresenter.class, e);
+        }
+
+        try {
+            debuggerDisconnectedChannel = JavaRuntimeExtension.DISCONNECT_CHANNEL + javaDebuggerInfo.getId();
+            messageBus.subscribe(debuggerDisconnectedChannel, debuggerDisconnectedHandler);
+        } catch (WebSocketException e) {
+            Log.error(JavaDebugger.class, e);
+        }
+    }
+
+    private void stopCheckingDebugEvents() {
+        try {
+            if (messageBus.isHandlerSubscribed(debuggerEventsHandler, debuggerEventsChannel)) {
+                messageBus.unsubscribe(debuggerEventsChannel, debuggerEventsHandler);
+            }
+
+            if (messageBus.isHandlerSubscribed(debuggerDisconnectedHandler, debuggerDisconnectedChannel)) {
+                messageBus.unsubscribe(debuggerDisconnectedChannel, debuggerDisconnectedHandler);
+            }
+        } catch (WebSocketException e) {
+            Log.error(JavaDebugger.class, e);
+        }
+    }
+
+    @Override
+    public Promise<String> getValue(String variable) {
+        if (!isConnected()) {
+            return Promises.reject(JsPromiseError.create("Debugger is not connected"));
+        }
+
+        Promise<Value> promise = service.getValue(javaDebuggerInfo.getId(), dtoFactory.createDtoFromJson(variable, Variable.class));
+        return promise.then(new Function<Value, String>() {
+            @Override
+            public String apply(Value arg) throws FunctionException {
+                List<Variable> variables = arg.getVariables();
+                return dtoFactory.toJson(variables);
+            }
+        }).catchError(new Operation<PromiseError>() {
+            @Override
+            public void apply(PromiseError arg) throws OperationException {
+                Log.error(JavaDebugger.class, arg.getMessage());
+            }
+        });
+    }
+
+    @Override
+    public Promise<String> getStackFrameDump() {
+        if (!isConnected()) {
+            return Promises.reject(JsPromiseError.create("Debugger is not connected"));
+        }
+
+        Promise<StackFrameDump> promise = service.getStackFrameDump(javaDebuggerInfo.getId());
+        return promise.then(new Function<StackFrameDump, String>() {
+            @Override
+            public String apply(StackFrameDump arg) throws FunctionException {
+                return dtoFactory.toJson(arg);
+            }
+        }).catchError(new Operation<PromiseError>() {
```

Comment:
Call catchError on method which called method getStackFrameDump(), but not here.